contracts/JBTieredNFTRewardDelegate.sol

```diff
@@ -0,0 +1,388 @@
+  error NOT_AVAILABLE();
+  error INVALID_IDS();
+
+  //*********************************************************************//
+  // --------------------- internal stored properties ------------------ //
+  //*********************************************************************//
+
+  /** 
+    @notice
+    The reward tiers. 
+  */
+  JBNFTRewardTier[] internal _tiers;
+
+  //*********************************************************************//
+  // --------------- public immutable stored properties ---------------- //
+  //*********************************************************************//
+
+  /** 
+    @notice
+    The token to expect contributions to be made in.
+  */
+  address public immutable override contributionToken;
+
+  //*********************************************************************//
+  // ------------------------- external views -------------------------- //
+  //*********************************************************************//
+
+  /** 
+    @notice 
+    The reward tiers.
+
+    @return The tiers. 
+  */
+  function tiers() external view override returns (JBNFTRewardTier[] memory) {
+    return _tiers;
+  }
+
+  /** 
+    @notice 
+    The total supply of issued NFTs from all tiers.
+
+    @return supply The total number of NFTs between all tiers.
+  */
+  function totalSupply() external view override returns (uint256 supply) {
+    // Keep a reference to the number of tiers.
+    uint256 _numTiers = _tiers.length;
+
+    // Keep a reference to the tier being iterated on.
+    JBNFTRewardTier memory _tier;
+
+    for (uint256 _i; _i < _numTiers; ) {
+      // Set the tier being iterated on.
+      _tier = _tiers[_i];
+
+      // Increment the total supply with the amount used already.
+      supply += _tier.initialAllowance - _tier.remainingAllowance;
+
+      unchecked {
+        ++_i;
+      }
+    }
+  }
+
+  /** 
+    @notice 
+    The total number of tokens with the given ID.
+
+    @return Either 1 if the token has been minted, or 0 if it hasnt.
+  */
+  function tokenSupply(uint256 _tokenId) external view override returns (uint256) {
+    return _ownerOf[_tokenId] != address(0) ? 1 : 0;
+  }
+
+  /** 
+    @notice 
+    The total number of tokens owned by the given owner. 
+
+    @param _owner The address to check the balance of.
+
+    @return The number of tokens owners by the owner.
+  */
+  function totalOwnerBalance(address _owner) external view override returns (uint256) {
+    return _balanceOf[_owner];
+  }
+
+  /** 
+    @notice 
+    The total number of tokens with the given ID owned by the given owner. 
+
+    @param _owner The address to check the balance of.
+    @param _tokenId The ID of the token to check the owner's balance of.
+
+    @return Either 1 if the owner has the token, or 0 if it does not.
+  */
+  function ownerTokenBalance(address _owner, uint256 _tokenId)
+    external
+    view
+    override
+    returns (uint256)
+  {
+    return _ownerOf[_tokenId] == _owner ? 1 : 0;
```

Comment:
Do we want/need to keep the above `ownerTokenBalance`, `totalOwnerBalance` and `tokenSupply` methods? We are extending solmate ERC721 so we already have public `balanceOf` and `ownerOf` methods